e2e/config/vhd.go

```diff
@@ -0,0 +1,176 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+)
+
+// VHD represents a VHD used to run AgentBaker E2E scenarios.
+type VHD struct {
+	ImageID         string
+	VersionTagName  string
+	VersionTagValue string
+	// ResourceID is the resource ID pointing to the underlying VHD in Azure. Based on the current setup, this will always be the resource ID
+	// of an image version in a shared image gallery.
+	resourceID               VHDResourceID
+	buildResourceID          VHDResourceID
+	fetchBuildResourceIDOnce sync.Once
+	fetchResourceIDOnce      sync.Once
+}
+
+var (
+	VHDUbuntu1804Gen2Containerd = &VHD{
+		ImageID:         "/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/1804Gen2",
```

Comment:
nit: can we declare the template string `/subscriptions/8ecadfc9-d1a3-4ea4-b844-0d9f87e4d7c8/resourceGroups/aksvhdtestbuildrg/providers/Microsoft.Compute/galleries/PackerSigGalleryEastUS/images/` as a constant? For easier maintenance purpose.

Reply:
I don't think it will improve maintainability. But it will add extra level of indirection.
When I navigate through the code, I like to be able to find the definition as soon as possible.
I also think it's easier to read when all things related to the image is in a single place and not scattered in the file.

Reply:
I was just thinking if one day we change the resource group or subscription, we can simply change 1 place instead of searching for all and replace. Anyway, just my own way. Feel free to keep yours🙂

Reply:
I generally try to avoid this approach (In some cases it makes more sense though). Find and replace is trivial operation in modern editors.
But trying to mentally execute a piece of code is an extra burden every reader will need to perform.

Reply:
I do somewhat agree with Devin here, at least the subscription ID, gallery name and image definition can be high level consts defined earlier in the file and then propagated downwards. for most of the cases, the first 2 don't even change across different definitions. 

but say in a case where we update the gallery name or move to a different subscriptions, this change is just harder to make.

Reply:
Not the hill I'm willing to die on. I removed the duplication.

But I think once in 3 years performing a single find and replace to change the subscription is trivial.
On other hand now it's not obvious anymore which image is used. And it's not searchable in the code anymore.